micronaut-camunda-bpm-feature/src/test/java/info/novatec/micronaut/camunda/bpm/feature/test/JobExecutorCustomizerTest.java

```diff
@@ -0,0 +1,30 @@
+package info.novatec.micronaut.camunda.bpm.feature.test;
+
+import io.micronaut.test.extensions.junit5.annotation.MicronautTest;
+import org.camunda.bpm.engine.ProcessEngine;
+import org.camunda.bpm.engine.ProcessEngineConfiguration;
+import org.camunda.bpm.engine.impl.cfg.ProcessEngineConfigurationImpl;
+import org.junit.jupiter.api.BeforeAll;
+import org.junit.jupiter.api.Test;
+
+import javax.inject.Inject;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+
+@MicronautTest
+public class JobExecutorCustomizerTest {
+
+
+    @Inject
+    private  ProcessEngine processEngine;
+
+    private  ProcessEngineConfigurationImpl processEngineConfigurationImpl;
+
+    @Test
+    void testJobExecutorCustomizer() {
+        processEngineConfigurationImpl = (ProcessEngineConfigurationImpl) processEngine.getProcessEngineConfiguration();
+
+        assertEquals(2000, processEngineConfigurationImpl.getJobExecutor().getWaitTimeInMillis());
```

Comment:
Unfortunately, this test is green - but its not working as expected. You are calling the callback too late.

If you start the application you see process instances being started every five seconds - but this should actually be every second according to the timer definition and with waittimeinmillis < 1000 we should be able to observe this.

To fix this implement in MnProcessEngineConfiguration:

```
@Override
    protected void initJobExecutor() {
        JobExecutor jobExecutor = new DefaultJobExecutor();
        jobExecutor.setWaitTimeInMillis(150);
        setJobExecutor(jobExecutor);
        super.initJobExecutor();
    }
```

This actually works.

However, we should also make the JobExecutor a bean and inject it into MnProcessEngineConfiguration to be able to set it in this method. And in the bean factory we should call JobExecutorCustomizer::customize before returning it.

Reply:
Done